hardware/diskinfo.py

```diff
@@ -43,43 +43,43 @@ def get_disk_info(name, sizes, hw_lst):
                  'queue/optimal_io_size', 'queue/physical_block_size',
                  'queue/rotational', 'queue/nr_requests']
     for info in info_list:
-        disk_sys_info = '/sys/block/%s/%s' % (name, info)
+        disk_sys_info = f'/sys/block/{name}/{info}'
         # revision can be explicitly named
         if not os.path.exists(disk_sys_info) and info == 'device/rev':
             info = 'device/revision'
         # for nvme devices we can have a nested device dir
         if not os.path.exists(disk_sys_info):
-            disk_sys_info = '/sys/block/%s/device/%s' % (name, info)
+            disk_sys_info = f'/sys/block/{name}/device/{info}'
         try:
-            with open(disk_sys_info, 'r') as dev:
+            with open(disk_sys_info) as dev:
                 hw_lst.append(('disk', name, info.split('/')[1],
                                dev.readline().rstrip('\n').strip()))
         except Exception as exc:
             sys.stderr.write(
-                'Failed retrieving disk information %s for %s: %s\n' % (
+                'Failed retrieving disk information {} for {}: {}\n'.format(
```

Comment:
Why not use formatted strings here too?

Reply:
no particular reason, this was a scripted change using pyupgrade
I'd also rather have more consistency so I'll change the rest to f-strings